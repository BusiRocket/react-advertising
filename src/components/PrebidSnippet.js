/**
 * PrebidSnippet.js
 *
 * (C) 2017 mobile.de GmbH
 *
 * @author <a href="mailto:pahund@team.mobile.de">Patrick Hund</a>
 * @since 09 Aug 2017
 */

import PropTypes from 'prop-types';
import React from 'react';

function PrebidSnippet({ active, scriptPath }) {
    const loadPrebidScriptTag = active ? <script async src={scriptPath} /> : null;
    return (
        <div>
            {loadPrebidScriptTag}
            <script
                dangerouslySetInnerHTML={{
                    __html: 'var pbjs=pbjs||{};pbjs.que=pbjs.que||[]'
                }}
            />
        </div>
    );
}

PrebidSnippet.propTypes = {
    active: PropTypes.bool,
    scriptPath: PropTypes.string.isRequired
};

export default PrebidSnippet;
