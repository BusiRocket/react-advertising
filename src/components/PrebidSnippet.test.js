/**
 * PrebidSnippet.test.js
 *
 * (C) 2017 mobile.de GmbH
 *
 * @author <a href="mailto:pahund@team.mobile.de">Patrick Hund</a>
 * @since 31 Aug 2017
 */
import React from 'react';
import PrebidSnippet from './PrebidSnippet';
import renderer from 'react-test-renderer';

describe('[@mt-advertising/components/PrebidSnippet]', () =>
    describe('The PrebidSnippet component', () => {
        describe('with just a mount point', () => snapshotTest(<PrebidSnippet scriptPath="path/to/my/script.js" />));
        describe('with active prop set', () =>
            snapshotTest(<PrebidSnippet scriptPath="path/to/my/script.js" active />));
    }));

function snapshotTest(component) {
    it('is rendered correctly', () => expect(renderer.create(component).toJSON()).toMatchSnapshot());
}
