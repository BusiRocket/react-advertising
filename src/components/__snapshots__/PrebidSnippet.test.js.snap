// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PrebidSnippet component with active prop set is rendered correctly 1`] = `
<div>
  <script
    async={true}
    src="path/to/my/script.js"
  />
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "var pbjs=pbjs||{};pbjs.que=pbjs.que||[]",
      }
    }
  />
</div>
`;

exports[`The PrebidSnippet component with just a mount point is rendered correctly 1`] = `
<div>
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "var pbjs=pbjs||{};pbjs.que=pbjs.que||[]",
      }
    }
  />
</div>
`;
